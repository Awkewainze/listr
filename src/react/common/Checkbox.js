import React from 'react';
import './Checkbox.scss';

const Checkbox = ({
  className = '',
  id,
  checked = false,
  onChange
}) => {
  return (
    <div className={`Checkbox ${className}`}>
      <input type="checkbox" className="Checkbox__hidden" id={id} checked={checked} onChange={onChange}/>
      <label htmlFor={id} className={`Checkbox__styled ${checked ? 'Checkbox__checked' : ''}`}>
        <svg className="Checkbox__icon" viewBox="0 0 24 24">
          <polyline points="20 6 9 17 4 12" />
        </svg>
      </label>
    </div>
  );
};

export default Checkbox;