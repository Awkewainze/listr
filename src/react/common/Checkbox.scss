.Checkbox {
  display: inline-block;
  vertical-align: middle;
  width: 16px;
  margin: 0 auto;

  .Checkbox__hidden {
    border: 0;
    clip: rect(0 0 0 0);
    clip-path: inset(50%);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    white-space: nowrap;
    width: 1px;

    &:focus + & {
      box-shadow: 0 0 0 3px pink;
    }
  }

  .Checkbox__styled {
    display: inline-block;
    width: 16px;
    height: 16px;
    background-color: papayawhip;
    border-radius: 3px;
    transition: all 150ms;
    cursor: pointer;

    &.Checkbox__checked {
      background-color: salmon;

      .Checkbox__icon {
        visibility: visible;
      }
    }

    .Checkbox__icon {
      fill: none;
      stroke: white;
      stroke-width: 2px;
      visibility: hidden;
    }
  }
}