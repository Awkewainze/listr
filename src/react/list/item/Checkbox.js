import React from 'react';
import './Checkbox.scss';

const Checkbox = ({
  className,
  checked,
  index,
  onChange
}) => (
  <div className={`Checkbox ${className}`}>
    <input className="Checkbox__toggle" type="checkbox" id={`Checkbox-${index + 1}`} onChange={onChange} checked={checked}/>
    <label className="Checkbox__toggleLabel" htmlFor={`Checkbox-${index + 1}`}/>
  </div>
);

export default Checkbox;