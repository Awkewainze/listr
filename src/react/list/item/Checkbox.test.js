import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './Checkbox';

const onChange = jest.fn();

it('matches snapshot', () => {
  expect(shallow(<Checkbox onChange={onChange} index={1} checked={false} />)).toMatchSnapshot();
});

it('is unchecked correctly', () => {
  const container = shallow(<Checkbox onChange={onChange} index={1} checked={false} />);
  expect(container.find('input[type="checkbox"]').prop('checked')).toBeFalsy();
});

it('is checked correctly', () => {
  const container = shallow(<Checkbox onChange={onChange} index={1} checked={true} />);
  expect(container.find('input[type="checkbox"]').prop('checked')).toBeTruthy();
});

it('calls onChange when checked or unchecked', () => {
  const expectedIndex = 2;
  const expectedEvent = new window.Event('change');

  const container = shallow(<Checkbox onChange={onChange} index={expectedIndex} checked={true} />);
  container.find('input[type="checkbox"]').simulate('change', expectedEvent);
  expect(onChange.mock.calls.length).toEqual(1);
  expect(onChange.mock.calls[0][0]).toEqual(expectedIndex);
  expect(onChange.mock.calls[0][1]).toEqual(expectedEvent);
});