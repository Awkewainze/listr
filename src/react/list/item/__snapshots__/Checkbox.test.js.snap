// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Checkbox undefined"
  onClick={[Function]}
>
  <input
    checked={false}
    className="Checkbox__toggle"
    id="Checkbox-2"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__toggleLabel"
    htmlFor="Checkbox-2"
  />
</div>
`;
